autotx/utils/agent/build_goal.py

```diff
@@ -66,35 +66,61 @@ def build_goal(prompt: str, agents_information: str, smart_account_address: ETHA
 def analyze_user_prompt(chat_history: str, agents_information: str, smart_account_address: ETHAddress) -> DefineGoalResponse:
     template = dedent(
         """
-        Based on the following chat history between you and the user: 
-        ```
-        {chat_history}
-        ```
-            
+        You are a task assignment expert. You are good at analyzing the user's chat records to complete the task goals and conditions represented in the user's chat records.
         You must analyze the prompt and define a goal to be executed by the agents.
         If the prompt is not clear or missing information, you MUST ask for more information.
         If the prompt is invalid, unsupported or outside the scope of the agents, you MUST ask for a new prompt.
         Always ensure you have all the information needed to define the goal that can be executed without prior context.
+        When dealing with Ethereum transactions, assume the following is the user's address: {smart_account_address}
         
         The available agents and tools:
         {agents_information}
 
         Respond ONLY in one of three of the following JSON formats:
-        1:
+        
         {{
             "type": "goal",
-            "goal": "The detailed goal here. No need to mention specific agents or tools. But you MUST mention the user's address."
+            "goal": "The detailed goal here. No need to mention specific agents or tools."
         }}
-        2:
+        
         {{
             "type": "missing_info",
             "message": "The information that is missing here"
         }}
-        3:
+        
         {{
             "type": "unsupported",
             "message": "Reason why the prompt is unsupported here"
         }}
+        
+        Please note: You can only output content in json format, and do not output any other content!
+        
+        You can refer to the following examples:
+        Chat History: User: Swap ETH to 0.05 WBTC, then swap WBTC to 1000 USDC, and finally send 50 USDC to vitalik.eth
+        Output: 
+        {{
+            "type": "goal",
+            "goal": "First, swap Ethereum (ETH) to 0.05 Wrapped Bitcoin (WBTC), ensuring the right amount of ETH is swapped for exactly 0.05 WBTC. After successfully swapping to WBTC, proceed to swap 0.05 WBTC to 1000 USD Coin (USDC), making sure to receive exactly 1000 USDC from the swap. Following these swaps, transfer 50 USDC to the Ethereum Name Service (ENS) domain 'vitalik.eth'."
```

Comment:
> proceed to swap 0.05 WBTC to 1000 USD Coin (USDC)

from my experience, this might confuse the LLM, making it try to swap the entire 0.05 WBTC to USDC rather than buying 1000 USDC with the WBTC (1000 USDC ~= 0.014 WBTC). This might come from our prompt examples (which need to be improved) but just in case :smile:

Reply:
Okay, I'll try to modify this